lib/matplotlib/widgets.py

```diff
@@ -2366,103 +2367,122 @@ def closest(self, x, y):
+        "box"``.  Default::
 
-        useblit : bool, default: False
-            Whether to use blitting for faster drawing (if supported by the
-            backend).
+            dict(facecolor="red", edgecolor="black", alpha=0.2, fill=True)
 
-        lineprops : dict, optional
-            Properties with which the line is drawn, if ``drawtype == "line"``.
-            Default::
+    spancoords : {"data", "pixels"}, default: "data"
+        Whether to interpret *minspanx* and *minspany* in data or in pixel
+        coordinates.
 
-                dict(color="black", linestyle="-", linewidth=2, alpha=0.5)
+    button : `.MouseButton`, list of `.MouseButton`, default: all buttons
+        Button(s) that trigger rectangle selection.
 
-        rectprops : dict, optional
-            Properties with which the rectangle is drawn, if ``drawtype ==
-            "box"``.  Default::
+    handle_grab_distance : float, default: 10
+        Distance in pixels within which the interactive tool handles can be
+        activated.
+
+    handle_props : dict
+        Properties with which the interactive handles are drawn.
 
-                dict(facecolor="red", edgecolor="black", alpha=0.2, fill=True)
+    interactive : bool, default: False
+        Whether to draw a set of handles that allow interaction with the
+        widget after it is drawn.
 
-        spancoords : {"data", "pixels"}, default: "data"
-            Whether to interpret *minspanx* and *minspany* in data or in pixel
-            coordinates.
+    state_modifier_keys : dict, optional
+        Keyboard modifiers which affect the widget's behavior.  Values
+        amend the defaults.
 
-        button : `.MouseButton`, list of `.MouseButton`, default: all buttons
-            Button(s) that trigger rectangle selection.
+        - "move": Move the existing shape, default: no modifier.
+        - "clear": Clear the current shape, default: "escape".
+        - "square": Makes the shape square, default: "shift".
+        - "center": Make the initial point the center of the shape,
+          default: "ctrl".
 
-        maxdist : float, default: 10
-            Distance in pixels within which the interactive tool handles can be
-            activated.
+        "square" and "center" can be combined.
 
-        marker_props : dict
-            Properties with which the interactive handles are drawn.  Currently
-            not implemented and ignored.
+    drag_from_anywhere : bool, optional
+        If `True`, the widget can be moved by clicking anywhere within
+        its bounds.
+    """
 
-        interactive : bool, default: False
-            Whether to draw a set of handles that allow interaction with the
-            widget after it is drawn.
 
-        state_modifier_keys : dict, optional
-            Keyboard modifiers which affect the widget's behavior.  Values
-            amend the defaults.
+@docstring.Substitution(_RECTANGLESELECTOR_PARAMETERS_DOCSTRING)
```

Comment:
Why this indirection?

Reply:
To be able to reuse the docstring in EllipseSelector and RectangleSelector.

Reply:
I see. I'm not a fan of too much indirection in the docs. Some duplication is ok.
The risk with indirection is that people start to write about Rectangle explicitly, which does not make sense for the Ellipse case. Conversly, the risk with duplicated docstrings is that they are not in sync (which IMHO is not the end of the world).

It's a tradeoff which I would decide in favor of the duplication here. But I'm not going to fight over it.

Reply:
I have put a placeholder for the artist type to be replaced by `ellipse` or `selector` to fix this issue.

> I even claim it's a mistake that `EllipseSelector` inherits from `RectangleSelector`.

I don't know about that, maybe for another PR! ;)